### ISSUE TITLE
BinData store/restore?

### ISSUE THREAD

**Issue by iliakan** [Automattic/mongoose#4457]
I have a model with a mixed field, like this:

```js
const schema = new Schema({
  key: String,
  value: {}
}
```

I store `value = { type: 'pdf', body: <Buffer 25 50 44...> }`.

Then, when I read the data back, I get:

```js
{
  type: 'pdf',
 body:
   Binary {
     _bsontype: 'Binary',
     sub_type: 0,
     position: 832248,
     buffer: <Buffer 25 50... > },
}
```

That does not work, I want to get a normal buffer to body, as I stored it.

In brief, I put one object and then another one is returned. 

Should I transform it to buffer?
Or do I do something wrong?

**Comment by vkarpov15** [comment]
This is expected behavior ATM, that's just what the underlying mongodb driver does. The driver does have an option to convert buffers back into buffers rather than 'binary' added here: https://github.com/mongodb/js-bson/pull/116 but one of the layers of the driver doesn't support passing the option in. I added a PR to be able to set this transparently https://github.com/christkv/mongodb-core/issues/135 . Until that's merged (specifically the first diff https://github.com/christkv/mongodb-core/pull/135/files#diff-0) nothing mongoose can do about it unfortunately.

**Comment by vkarpov15** [comment]
Should be fixed in 4.6.4
